controllers/storageautoscaler/util.go

```diff
@@ -0,0 +1,124 @@
+	secrets := "/var/run/secrets/kubernetes.io/serviceaccount"
+	caCertPath := secrets + "/service-ca.crt"
+	tokenPath := secrets + "/token"
+
+	// create auth roundtripper
+	tlsConfig, err := prometheusconfig.NewTLSConfig(&prometheusconfig.TLSConfig{
+		CAFile: caCertPath,
+	})
+	if err != nil {
+		log.Error(err, "failed to create tls config")
+		return nil, err
+	}
+	settings := prometheusconfig.TLSRoundTripperSettings{}
+	newRT := func(cfg *tls.Config) (http.RoundTripper, error) {
+		return &http.Transport{TLSClientConfig: cfg}, nil
+	}
+	tlsRoundTripper, err := prometheusconfig.NewTLSRoundTripperWithContext(ctx, tlsConfig, settings, newRT)
+	if err != nil {
+		log.Error(err, "failed to create tls round tripper")
+		return nil, err
+	}
+	roundTripper := prometheusconfig.NewAuthorizationCredentialsRoundTripper("Bearer", prometheusconfig.NewFileSecret(tokenPath), tlsRoundTripper)
+	// create prometheus client
+	client := &http.Client{
+		Transport: roundTripper,
+		Timeout:   clientTimeout,
+	}
+
+	return client, nil
+}
+
+type PrometheusResponse struct {
+	Status string       `json:"status"`
+	Data   VectorResult `json:"data"`
+}
+
+// VectorResult contains only the vector result type
+type VectorResult struct {
+	ResultType string       `json:"resultType"`
+	Result     model.Vector `json:"result"`
+}
+
+func queryMetrics(client *http.Client, operatorNamespace string, query string, log logr.Logger) (model.Vector, error) {
+	prometheusURL, err := getPrometheusURL(operatorNamespace)
+	if err != nil {
+		log.Error(err, "failed to determine if ROSA HCP cluster")
+		return nil, err
+	}
+	queryUrl := fmt.Sprintf("%s/api/v1/query?query=%s", prometheusURL, url.QueryEscape(query))
+
+	resp, err := client.Get(queryUrl)
+	if err != nil {
+		log.Error(err, "error creating get request")
+		return nil, err
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		log.Error(err, "failed to read get response")
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var promResp PrometheusResponse
+	if err := json.Unmarshal([]byte(body), &promResp); err != nil {
+		log.Error(err, "error unmarshaling prometheusResponse JSON")
+		return nil, err
+	}
+	vector := promResp.Data.Result
+
+	return vector, nil
+}
+
+func scrapeMetrics(ctx context.Context, operatorNamespace string, query string, log logr.Logger) (model.Vector, error) {
+	client, err := getClient(ctx, log)
+	if err != nil {
+		log.Error(err, "failed to get scraper")
+		return nil, err
+	}
+	return queryMetrics(client, operatorNamespace, query, log)
+}
+
+func (r *StorageAutoscalerReconciler) updateStatus(ctx context.Context, storageAutoScaler, originalStorageAutoScaler *ocsv1.StorageAutoScaler) error {
+	if err := r.Status().Patch(ctx, storageAutoScaler, client.MergeFrom(originalStorageAutoScaler)); err != nil {
+		r.Log.Error(err, "failed to update storage autoscaler status")
+		return err
+	}
+	r.Log.Info("storage autoscaler status updated")
+	return nil
+}
```

Comment:
Doesn't look like it belongs here. Hence, I am against files named util that quickly becomes a dumping ground for functions.